aten/src/ATen/UndefinedTensor.h

```diff
@@ -7,6 +7,7 @@ namespace at {
 struct AT_API UndefinedTensor final : public TensorImpl {
 public:
   static inline UndefinedTensor * singleton() {
+    static UndefinedTensor _singleton;
```

Comment:
err, I originally wrote it like this, but @colesbury convinced me the other way was better because the runtime doesn't need to check if _singleton has been initialized on each call to singleton().

Reply:
The hazard of doing it this way is that now the constructor for `UndefinedTensor` gets invoked at initialization time, which sets you up for a possible static initializer order fiasco. Fortunately the Context does seem to be initialize on use, which prevents us from being hosed here, so I guess it should be OK.